### DISCUSSION TITLE
Login and logout toggle

### DISCUSSION THREAD

**Discussion by jjgonc** [directus/directus#6974] [Archive]
Hey, I'm working on the login, logout and register system of my frontend with **vuejs 3**, using the Directus JS SDK.
I would like to present the login button when no user is logged and the logout button when a user is logged, but I'm struggling a bit to do it, since I'm starting to get into this "world"...

Can I get some help pls?
Thanks!

**Comment by rijkvanzanten** [comment]
I'd recommend hitting the `/users/me` endpoint on initial load to check if the user is logged in, and to retrieve what user the current user is. Once you have that information, you can control whether or not to render the login form 👍🏻

**Reply by jjgonc** [reply]
Can you explain better what do you mean by "hitting `/users/me` endpoint on initial load" or what is the correspondant method from the JS SDK? 

I tried to place it in the `created` lifecycle hook, but when I don't have a user logged in, it returns a 401 (obviously it has no user to access and gives an error) and it doesn't toggle the isLogged boolean (tells if a user is logged or not and decides if the Login button appears or if the Logout button appears))... 

One way I thought would solve is if there is a method that recognizes an error (an if there is one I'd toggle isLogged to false), but I don't know if it does exist. Here is the code and the error I get when no user is logged

![Screenshot from 2021-07-27 10-45-33](https://user-images.githubusercontent.com/61986130/127133564-563f6c2b-4444-41ba-93c4-f3514d38caf8.png)
![Screenshot from 2021-07-27 10-46-15](https://user-images.githubusercontent.com/61986130/127133668-fba86af8-22b3-4b83-888f-fa2edf536418.png)



**UPDATE:**
I tried like this, but I have to F5 the page to make the login disappear and give the place to the logout... maybe it isn't like this what you were talking about :(

![Screenshot from 2021-07-27 11-18-14](https://user-images.githubusercontent.com/61986130/127138215-35a28a5b-34ac-4e05-acb4-45da75352cae.png)


Thx for the help!

**Reply by rijkvanzanten** [reply]
I don't think you can make `created` async in Vue, that's why you have to refresh before it works.

Try doing something like this!

```vue
<template>
  <p v-if="loading">Loading...</p>
  <p v-else-if="loggedIn">I'm Logged In!</p>
  <p v-else>You need to login...</p>
</template>

<script>
import directus from '../my-directus';

export default {
  data() {
     return {
        loading: false,
        loggedIn: false,
     }
  },
  created() {
     checkLogin()
  },
  methods: {
     async checkLogin() {
        this.loading = true;

        try {
           await directus.users.me();
           this.loggedIn = true;
        } catch {
           this.loggedIn = false;
        } finally {
           this.loading = false;
        }
     }
  }
}
</script>
```

**Reply by jjgonc** [reply]
I tried this solution, but seems like it isn't possible to call the checkLogin() like that (it displays this error and opens an all blank page)

![Screenshot from 2021-07-27 16-29-54](https://user-images.githubusercontent.com/61986130/127182771-0c23261d-292e-44d2-a136-766e46773656.png)

**Reply by rijkvanzanten** [reply]
Oh yeah, that `createLogin` needs to be `this.createLogin()` 👍🏻

**Reply by jjgonc** [reply]
> Oh yeah, that `createLogin` needs to be `this.createLogin()` 👍🏻

also tried that, but no success :(
